app/src/features/mocks/modals/ExportMocksModalWrapper/ExportMocksModal.tsx

```diff
@@ -0,0 +1,163 @@
+import React, { useEffect, useState } from "react";
+import { RQModal } from "lib/design-system/components";
+import { HiOutlineShare } from "@react-icons/all-files/hi/HiOutlineShare";
+import { PiWarningCircleBold } from "@react-icons/all-files/pi/PiWarningCircleBold";
+import { Button, Spin } from "antd";
+import { RQMockSchema } from "components/features/mocksV2/types";
+import { getMock } from "backend/mocks/getMock";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/selectors";
+import { getCurrentlyActiveWorkspace } from "store/features/teams/selectors";
+import { prepareMocksToExport } from "./utils";
+import fileDownload from "js-file-download";
+import { getFormattedDate } from "utils/DateTimeUtils";
+import { toast } from "utils/Toast";
+import { trackMocksExported } from "modules/analytics/events/features/mocksV2";
+import Logger from "lib/logger";
+import "./ExportMocksModal.scss";
+
+const EmptySelectionView = () => {
+  return (
+    <div className="sharing-modal-empty-view sharing-modal-body">
+      <PiWarningCircleBold />
+      <div className="title text-white text-bold">Please select the mocks that you want to share!</div>
+    </div>
+  );
+};
+
+interface ShareMocksModalProps {
+  isOpen: boolean;
+  toggleModal: () => void;
+  selectedMockIds: string[];
+  onSuccess?: () => void;
+}
+
+export const ExportMocksModal: React.FC<ShareMocksModalProps> = ({
+  isOpen,
+  toggleModal,
+  selectedMockIds,
+  onSuccess = () => {},
+}) => {
+  const user = useSelector(getUserAuthDetails);
+  const workspace = useSelector(getCurrentlyActiveWorkspace);
+  const uid = user?.details?.profile?.uid;
+  const teamId = workspace?.id;
+
+  const [isMocksLoading, setIsMocksLoading] = useState(false);
+  const [mocks, setMocks] = useState<RQMockSchema[]>([]);
+  const [mocksExportDetails, setMocksExportDetails] = useState<{
+    fileContent: string;
+    mocksCount: number;
+    collectionsCount: number;
+  }>(null);
```

Comment:
I am sorry, I was not aware `null` is a valid type for this state when I wrote [this comment](https://github.com/requestly/requestly/pull/2123#discussion_r1740723906)

can you please bring back the old change

Reply:
Not needed, keeping it as it is